src/main/java/org/spongepowered/api/event/item/merchant/TradeWithMerchantEvent.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.item.merchant;
+
+import org.spongepowered.api.entity.living.Humanoid;
+import org.spongepowered.api.event.Cancellable;
+import org.spongepowered.api.event.Event;
+import org.spongepowered.api.item.merchant.Merchant;
+import org.spongepowered.api.item.merchant.TradeOffer;
+
+/**
+ * An event when a {@link Humanoid} trades with a {@link Merchant}.
+ */
+public interface TradeWithMerchantEvent extends Event, Cancellable {
+
+	/**
+	 * Gets the {@link Merchant} which is trading.
+	 *
+	 * @return The trading merchant.
+	 */
+	Merchant getMerchant();
+
+	/**
```

Comment:
It would be nice to have the original TradeOffer here.

Reply:
I'm not sure whether the `TradeOffer` should be settable in this event.

Reply:
@ST-DDT Would you care to share with the class why?

Reply:
It's hard to explain. I dont see a reason for that. You are in the middle of the trade. Why would the trade change at this point. If this were something like `SelectTradeOfferEvent` it would make sense, but not while actually doing it.

Reply:
Considering that case you mentioned, I am not too sure if you can make the client forcefully select a TradeOffer. Maybe it would be best to make the TradeOffer not settable during the event.